### PR TITLE
Create a cmake export target

### PR DESCRIPTION
**Pull request by gillins**
And use it when building the Python bindings.

Should make it easier for `cmake` projects to link against `libkea`.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -54,7 +54,17 @@ set_target_properties(${LIBKEA_LIB_NAME}
...
+  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>
+  $<INSTALL_INTERFACE:include>
+)
+# TODO: a better way??
+if(MSVC)
+    target_link_libraries(Kealib INTERFACE "$<INSTALL_INTERFACE:${CMAKE_INSTALL_PREFIX}/lib/libkea${CMAKE_IMPORT_LIBRARY_SUFFIX}>")
+else()
+    target_link_libraries(Kealib INTERFACE "$<INSTALL_INTERFACE:${CMAKE_INSTALL_PREFIX}/lib/libkea${CMAKE_SHARED_LIBRARY_SUFFIX}>")
+endif(MSVC)
```
Comment (gillins) [inline src/CMakeLists.txt:67]: If anyone knows why I have to do all this I'd appreciate a tidier way!

**Comment by gillins** [comment]
> Given the project name and the Kealib target name, it might have more sense to create Kealib CMake package with a Kealib::Kealib target.

Agreed. Perhaps we can change this for 1.6.0 (I'm hoping to push out 1.5.3 soon)? 1.6.0 will have some larger changes so that would be a good time to rename.

**Comment by dg0yt** [comment]
It might be possible to have both in 1.5.3, and then remove the other one in 1.6.0.

**Comment by dg0yt** [comment]
I can do that. With your indication of acceptance, I can test this in the vcpkg port as part of a fix which includes a usage synopsis. And I would prefer to never document anything else ;-)

However, to minimize impact, I suggest two steps:
- For 1.5.3: Leave export mostly as is, and generate the Kealib config/namespace as a separate trivial config file / interface target.
  This will leave all existing targets where they are now.
- for 1.6.0: Get rid of all `libkea` config/namespace stuff, and move to Kealib-only.

**Comment by gillins** [comment]
Sounds good. We'd also have to submit something to GDAL that looks for the new target, but can also try the old one (for people still on older kealib). Not sure if this is possible?

**Comment by dg0yt** [comment]
> We'd also have to submit something to GDAL that looks for the new target, but can also try the old one (for people still on older kealib). Not sure if this is possible?

This is where I come from: https://github.com/microsoft/vcpkg/pull/35437/files
Once Kealib's CMake config is stable, we can look at a GDAL PR.

**Pull request merged**